src/mm-input/mm-input.js

```diff
@@ -4,213 +4,166 @@
  * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
 
 */
-Polymer('mm-input', {
-	ver:"<<version>>",
-	//defaults
-	STATE_OPENED: "opened",
-	STATE_CLOSED: "closed",
-	PRIMARY_ICON_COLOR: Colors.A18,
-	PADDING_ICON: 25,
-
-	searchItems:[],
-	state: "closed",
-	placeholder:"",
-	validation:"none",
-	overflow: "hidden", // default width takes precedence
-	type: "text", // input default type
-	icon: "search", // default icon type 
-	search: false,
-	clear: false,
-	error: false,
-	autoItemSelected: false,
-	autocomplete: false,
-	paddingRight: 10,
-
-	publish: {
-		value: { value: "", reflect: true },
-		disabled: { value: false, reflect: true },
-		fitparent: { value: false, reflect: true },
-		layout: { value: null, reflect: true },
-		data: []
-	},
-
-	ready: function() {
-		this.email = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-		this.alpha = /^\w+$/;
-		this.inter = /^\d+$/;
-		this.floater = /^\d*[.]\d+$/;
-	},
-
-	dataChanged: function(oldData, newData) {
-		if (typeof newData === "string") {
-			try {
-				this.data = JSON.parse(newData);
-			} catch(e) {
-				this.data = newData;
+(function (scope) {
+
+	scope.Input = Polymer({
+		is: 'mm-input',
+
+		behaviors: [
+			StrandTraits.Stylable,
+			StrandTraits.Validate
+		],
+
+		properties: {
+			PRIMARY_ICON_COLOR: {
+				type:String, 
```

Comment:
PRIMARY_ICON_COLOR is used in a binding - so yes - it has to be published. Moving the others.